icechunk-python/src/storage.rs

```diff
@@ -51,62 +64,44 @@ impl PyS3Credentials {
 }
 
 #[pyclass(name = "StorageConfig")]
-pub enum PyStorageConfig {
-    Memory {
-        prefix: Option<String>,
-    },
-    Filesystem {
-        root: String,
-    },
-    S3 {
-        bucket: String,
-        prefix: String,
-        anon: bool,
-        credentials: Option<PyS3Credentials>,
-        endpoint_url: Option<String>,
-        allow_http: Option<bool>,
-        region: Option<String>,
-    },
-}
+pub struct PyStorageConfig(StorageConfig);
 
 #[pymethods]
 impl PyStorageConfig {
     #[classmethod]
     #[pyo3(signature = (prefix = None))]
     fn memory(_cls: &Bound<'_, PyType>, prefix: Option<String>) -> Self {
-        PyStorageConfig::Memory { prefix }
+        Self(StorageConfig::InMemory { prefix })
     }
 
     #[classmethod]
-    fn filesystem(_cls: &Bound<'_, PyType>, root: String) -> Self {
-        PyStorageConfig::Filesystem { root }
+    fn filesystem(_cls: &Bound<'_, PyType>, root: PathBuf) -> Self {
+        Self(StorageConfig::LocalFileSystem { root })
     }
 
     #[classmethod]
     #[pyo3(signature = (
         bucket,
         prefix,
         endpoint_url = None,
-        allow_http = None,
+        allow_http = false,
```

Comment:
`allow_http` can only be a bool, so why not default to `false` instead of defaulting to `None` and then unwrapping?

Reply:
no reason, just moving fast